isthmus/src/test/java/io/substrait/isthmus/ExpressionConvertabilityTest.java

```diff
@@ -78,6 +79,54 @@ public void singleOrList() throws IOException {
                     R.BOOLEAN,
                     b.fieldReference(commonTable, 0),
                     b.i32(10)))
-            .accept(to));
+            .accept(to),
+        expression.accept(to));
+  }
+
+  @Test
+  public void switchExpression() throws IOException {
+    Plan.Root root =
+        b.root(
+            b.filter(
+                input ->
+                    b.switchExpression(
+                        b.fieldReference(input, 0),
+                        List.of(
+                            b.switchClause(b.i32(5), b.i32(1)),
+                            b.switchClause(b.i32(10), b.i32(2))),
+                        b.i32(3)),
+                commonTable));
+    var relNode = converter.convert(root.getInput());
+    var expression =
+        ((Filter) relNode)
+            .getCondition()
+            .accept(
+                new RexExpressionConverter(
+                    CASE,
+                    new ScalarFunctionConverter(
+                        SimpleExtension.loadDefaults().scalarFunctions(), typeFactory)));
+    var to = new ExpressionProtoConverter(new ExtensionCollector(), null);
+    assertEquals(
+        b.ifThen(
+                List.of(
+                    b.ifClause(
+                        b.scalarFn(
+                            "/functions_comparison.yaml",
+                            "equal:any_any",
+                            R.BOOLEAN,
+                            b.fieldReference(commonTable, 0),
+                            b.i32(5)),
+                        b.i32(1)),
+                    b.ifClause(
+                        b.scalarFn(
+                            "/functions_comparison.yaml",
+                            "equal:any_any",
+                            R.BOOLEAN,
+                            b.fieldReference(commonTable, 0),
+                            b.i32(10)),
+                        b.i32(2))),
+                b.i32(3))
+            .accept(to),
+        expression.accept(to));
```

Comment:
It took me a while to figure out what this test is doing. If I understand this correctly, we can't just do a roundtrip test here because Calcite changes case switch statement to a series of case when expressions. What you're doing is:
1. Build a substrait plan with the case expression.
2. Convert it to Calcite
3. Pull out the expression from the filter.
4. Manually create the expected substrait equivalent as the first argument to the assert.
5. Compare the protobuf forms of both expressions.

Could you pull out 4 and assigned it to a variable like `expectedExpression`, along with a comment indicating how Calcite is changing the original expression to a new equivalent.

Reply:
Have a much better idea for this which I'm implementing.